clippy_lints/src/methods/slow_manual_contains.rs

```diff
@@ -0,0 +1,153 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::msrvs::{Msrv, SLICE_CONTAINS_SIMD};
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::{eq_expr_value, peel_hir_pat_refs};
+use rustc_ast::UnOp;
+use rustc_errors::Applicability;
+use rustc_hir::def::Res;
+use rustc_hir::{BinOpKind, Body, Expr, ExprKind, HirId, QPath};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self};
+use rustc_span::Span;
+use rustc_span::source_map::Spanned;
+
+use super::SLOW_MANUAL_CONTAINS;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, closure_arg: &Expr<'_>, msrv: &Msrv) {
+    let mut app = Applicability::MachineApplicable;
+
+    if !expr.span.from_expansion()
+        // check if `iter().any()` can be replaced with `contains()`
+        && let ExprKind::Closure(closure) = closure_arg.kind
+        && let Body{params: [param],value} = cx.tcx.hir().body(closure.body)
+        && let ExprKind::Binary(op, lhs, rhs) = value.kind
+        && let (peeled_ref_pat, _)=peel_hir_pat_refs(param.pat)
+        && let (true, snip)=can_replace_with_contains(cx, op, lhs, rhs, peeled_ref_pat.hir_id, &mut app)
```

Comment:
We should implement a `manual_option` lint :-)

More seriously, `can_replace_with_contains()` should return an `Option<String>`.